db/models.py

```diff
@@ -50,3 +59,48 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(default=datetime.now, blank=True)
+    user = models.ForeignKey(to=User, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ['-created_at']
+
+    def __str__(self):
+        return f"{self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = ForeignKey(to=MovieSession, on_delete=models.CASCADE)
+    order = ForeignKey(to=Order,on_delete=models.CASCADE)
+    row = IntegerField()
+    seat = IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["movie_session", "row", "seat"], name="unique ticket"),
+        ]
+
+    def __str__(self):
+        return f"{self.movie_session} (row: {self.row}, seat: {self.seat})"
+
+    def clean(self):
+        if self.row < 1 or self.row > self.movie_session.cinema_hall.rows:
+            raise ValidationError({
+                'row': [f"row number must be in available range: (1, rows): (1, {self.movie_session.cinema_hall.rows})"]
+            })
+        if self.seat < 1 or self.seat > self.movie_session.cinema_hall.seats_in_row:
+            raise ValidationError({
+                'seat': [
+                    f"seat number must be in available range: (1, seats_in_row): (1, {self.movie_session.cinema_hall.seats_in_row})"]
+            })
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        return super().save(*args, **kwargs)
```

Comment:
Ensure that the `movie_session` field is set before calling `full_clean()` in the `save` method. This is important because the `clean` method relies on `movie_session` to validate `row` and `seat` fields.